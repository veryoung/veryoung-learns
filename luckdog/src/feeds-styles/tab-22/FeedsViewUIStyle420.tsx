/**
 * @Author: veryoungwan
 * 排行榜外露卡片
 */

import React from 'react';
import { View } from '@tencent/hippy-react-qb';
import { vectorToArray } from './components/utils';
import { Title } from './components';
import FeedsSpliter from '../../components/FeedsSpliter';
import FeedsViewItem from '../FeedsViewItem';
import FeedsProtect from '../../mixins/FeedsProtect';
import { reportUDS, strictExposeReporter, BusiKey } from '@/luckdog';
import Rank from './components/Rank';
import { CommonCardStyle } from '../../framework/FeedsConst';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';
import { isTopTab } from '@/luckbox';
import { CommonProps } from '../../entity';

interface Props extends CommonProps {
  index: number;
}

@FeedsProtect.protect
export default class FeedsViewUIStyle420 extends FeedsViewItem<Props> {
  public static getRowType() {
    return 420;
  }
  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'FeedsViewUIStyle420');
  }

  public doBeaconByClick = (data = {}) => {
    reportUDS(BusiKey.CLICK__CARD, this.props, data);
  };

  public doBeaconByClickAll = () => {
    reportUDS(BusiKey.CLICK__CARD_VIEW_ALL, this.props);
  };

  public doBeaconBySlide = (moreData = {}) => {
    reportUDS(BusiKey.SLIDE__CARD, this.props, moreData);
  };

  public onTitleLayout = (event) => {
    strictExposeReporter.updateTitleHeight(this.props.index, event.layout.height);
  };

  public render() {
    countReRender(this, 'FeedsViewUIStyle420');
    const { itemBean = {}, index } = this.props;
    const { globalConf } = this.props;
    const { title = '', parsedObject = {} } = itemBean;
    const books = vectorToArray(parsedObject.vDetailData) || [];
    const titleRight = vectorToArray(parsedObject.vTextLink)[0] || [];

    return (
      <View
        style={{
          ...!isTopTab() && CommonCardStyle,
        }}
      >
        <FeedsSpliter style={globalConf.style} lineStyle={itemBean.bottomLineStyle} />
        <View>
          <Title
            title={title}
            parent={this}
            onLayout={this.onTitleLayout}
            right={titleRight}
            doBeaconByClick={this.doBeaconByClickAll}
          />
          <Rank
            parent={this}
            books={books}
            cardIndex={index}
            itemBean={itemBean}
            doBeaconByClick={this.doBeaconByClick}
            doBeaconBySlide={this.doBeaconBySlide}
          />
        </View>
      </View>
    );
  }
}
