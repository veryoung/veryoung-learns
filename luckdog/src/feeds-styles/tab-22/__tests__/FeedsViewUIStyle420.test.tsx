import React from 'react';
import renderer from 'react-test-renderer';

import FeedsViewUIStyle420 from '../FeedsViewUIStyle420';
import { commonProps } from '@test/mocks/props';

describe('<FeedsViewUIStyle420/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
        index: 0,
      };
      const tree = renderer.create(<FeedsViewUIStyle420 {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
