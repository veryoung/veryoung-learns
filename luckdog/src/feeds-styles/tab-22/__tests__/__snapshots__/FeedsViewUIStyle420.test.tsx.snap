// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsViewUIStyle420/> render testing render normal 1`] = `
<div
  nativeName="View"
  style={Object {}}
>
  <div
    nativeName="View"
    style={
      Array [
        Object {
          "height": 0,
        },
      ]
    }
  />
  <div
    nativeName="View"
    style={Object {}}
  >
    <div
      nativeName="View"
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        }
      }
    >
      <p
        nativeName="Text"
        numberOfLines={1}
        style={
          Array [
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "maxWidth": 252,
              "paddingBottom": 14,
              "paddingTop": 14,
            },
            Object {},
          ]
        }
        text=""
      >
        
      </p>
    </div>
    <div
      contentContainerStyle={
        Object {
          "paddingBottom": 16,
          "paddingHorizontal": 12,
        }
      }
      horizontal={true}
      nativeName="ScrollView"
      onScroll={[Function]}
      onScrollEndDrag={[Function]}
      scrollEventThrottle={100}
      sendMomentumEvents={true}
      showsHorizontalScrollIndicator={false}
      style={
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        }
      }
    >
      <div
        nativeName="View"
        style={
          Array [
            Object {
              "collapse": false,
              "flexDirection": "row",
            },
            Object {
              "paddingBottom": 16,
              "paddingHorizontal": 12,
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
